### PR TITLE
Implement pluggable time functions

### PR DESCRIPTION
**Pull request by jsumners**
This PR implements the time function plugin system discussed in #202 . It also removes the `slowtime` option.  Thus, this should necessitate a major version bump.

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -38,13 +42,12 @@
...
-  * `slowtime` (boolean): Outputs ISO time stamps (`'2016-03-09T15:18:53.889Z'`)
-    instead of Epoch time stamps (`1457536759176`). **WARNING**: This option
-    carries a 25% performance drop. We recommend using default Epoch timestamps and transforming logs after if required. The `pino -t` command will do this for
-    you (see [CLI](cli.md)). Default: `false`.
+  * `timestamp` (boolean|function): Enables or disables the inclusion of a timestamp in the
+    log message. If a function is supplied, it must synchronously return a JSON string
+    representation of the time, e.g. `"time":1493426328206 (which is the default).
+    If set to `false`, no timestamp will be included in the output.
+    See [stdTimeFunctions](#stdTimeFunctions) for more information.
+    Caution: any sort of formatted time will significantly slow down Pino's performance.
```
Comment (mcollina) [inline docs/API.md]: it's not exactly clear that a user should pass the functions defined in the pino modules.

**Comment by jsumners** [comment]
@mcollina note that @davidmarkclements wanted to deprecate `slowtime` -- https://github.com/pinojs/pino/issues/202#issuecomment-285053264 . I don't think it's worth keeping around for some indeterminate amount of time, which will likely lead to it never getting removed. So I went ahead and fully deprecated it.

File: docs/API.md (hunk 1 above)
Comment (jsumners) [inline docs/API.md]: I updated this text in 7455cd9

**Comment by davidmarkclements** [comment]
we could output a deprecation notice, and remove in a the next major

**Comment by mcollina** [comment]
Given the amount of churn needed for a semver-major release, I think the best option is to deprecate the option, but keep supporting it until we have to bump for some other reason.
We have a lot of dependents now, so we should be more careful than in the past.

What do the others thinks? I'm ok if you all want to bump the major: I just want to be lazy and not update my deps

**Comment by davidmarkclements** [comment]
Do agree

**Comment by jsumners** [comment]
I have added the `slowtime` option back in. It emits a deprecation warning when used. I also created a project for tracking such things for v5.0.0 -- https://github.com/pinojs/pino/projects/1

**Pull request merged**